client/components/UserDecalPanel.js

```diff
@@ -87,6 +90,37 @@ class UserDetailsPanel extends Component {
     return null
   }
 
+  _renderDecalCode = (decal, index) => (
+    <>
+      <div className="decal-code" key={`code${index}`}>
```

Comment:
You should never use `index` in a key attribute. 
Since we are mapping out a decal resource, we should use it's id

```suggestion
      <div className="decal-code" key={decal.id}>
```